integ-tests/src/cucumberTest/java/cucumber/steps/dataservice/DataServiceSteps.java

```diff
@@ -194,48 +191,28 @@ public DataServiceSteps() {
+          assertThat(exceptionContext.getException())
+              .isNotNull()
+              .satisfies(t -> assertThat(t.getClass().getSimpleName()).isEqualTo(exceptionName));
         });
 
     And(
-        "^the resourceType is \"([^\"]*)\"$",
-        (final String resourceType) -> {
-          assertNotNull("Expecting an exception to be thrown", exceptionContext.getException());
-          if (exceptionContext.getException() instanceof ResourceNotFoundException) {
-            final ResourceNotFoundException exception =
-                (ResourceNotFoundException) exceptionContext.getException();
-            assertEquals(resourceType, exception.getResourceType());
-          } else if (exceptionContext.getException() instanceof ResourceExistsException) {
-            final ResourceExistsException exception =
-                (ResourceExistsException) exceptionContext.getException();
-            assertEquals(resourceType, exception.getResourceType());
-          } else if (exceptionContext.getException() instanceof ResourceInUseException) {
-            final ResourceInUseException exception =
-                (ResourceInUseException) exceptionContext.getException();
-            assertEquals(resourceType, exception.getResourceType());
-          }
+        "^its ([^ ]*) is \"([^\"]*)\"$",
+        (final String field, final String value) -> {
+          assertThat(exceptionContext.getException()).hasFieldOrPropertyWithValue(field, value);
         });
 
     And(
-        "^the resourceId contains \"([^\"]*)\"$",
-        (final String resourceId) -> {
-          assertNotNull("Expecting an exception to be thrown", exceptionContext.getException());
-          if (exceptionContext.getException() instanceof ResourceNotFoundException) {
-            final ResourceNotFoundException exception =
-                (ResourceNotFoundException) exceptionContext.getException();
-            assertEquals(resourceId, exception.getResourceType());
-          } else if (exceptionContext.getException() instanceof ResourceExistsException) {
-            final ResourceExistsException exception =
-                (ResourceExistsException) exceptionContext.getException();
-            assertEquals(resourceId, exception.getResourceType());
-          } else if (exceptionContext.getException() instanceof ResourceInUseException) {
-            final ResourceInUseException exception =
-                (ResourceInUseException) exceptionContext.getException();
-            assertEquals(resourceId, exception.getResourceType());
-          }
+        "^its ([^ ]*) contains \"([^\"]*)\"$",
+        (final String field, final String value) -> {
+          assertThat(exceptionContext.getException())
+              .hasFieldOrProperty(field)
+              .extracting(field)
+              .allSatisfy(s -> assertThat(s).asString().contains(value));
         });
 
+    And("^the resourceId contains \"([^\"]*)\"$", (final String resourceId) -> {});
```

Comment:
Looks like we already have the above step on "^its ([^ ]*) contains \"([^\"]*)\"$", do we still need this specific step on "^the resourceId contains \"([^\"]*)\"$" ?

Reply:
Oh whoops, yes this should be removed.